### ISSUE TITLE
Slow for rotated monitors

### ISSUE THREAD

**Issue by FortranFlo** [liskin/hybrid-screenclone#7]
Hi, and many thanks for programming and sharing this. Works great for my thinkpad, and 
I can connect two external monitors without problems.

I can even set up the second monitor to be rotated: set mode size of virtual display to
the total needed resolution in xrandr, then rotate the second monitor using 

xrandr -d :8 --output DP-3 --auto --output VGA-0 --auto --right-of DP-3 --rotate left

(xrandr -d :8 tells me there are two monitors connected at DP-3 (display port) and VGA-0)

Everything works very smooth if I do not rotate. 
If I rotate, things are correct, but the rotated display is very laggy. I do not know why, perhaps 
the damage events are not updated correctly?

Any input would be appreciated!

**Comment by liskin** [comment]
I am not sure if this is it, but the image copy operation is very slow if the image width is not a multiple of 16. All standard resolutions are, but if you rotate, it is the height that must be divisible by 16. Is it?

**Comment by FortranFlo** [comment]
Thank you for the quick reply!
Should be, it's a 1024x1280 = 16_(64x80), so the total screen is 2704x1280 = 16_(169x80)

Prompt:~$ xrandr -d :8                                [removed irrelevant lines]
Screen 0: minimum 8 x 8, current 2704 x 1280, maximum 16384 x 16384
VGA-0 connected 1024x1280+1680+0 left (normal left inverted right x axis y axis) 376mm x 301mm
   1280x1024      60.0*+   75.0  

DP-3 connected 1680x1050+0+0 (normal left inverted right x axis y axis) 473mm x 296mm
   1680x1050      60.0*+

Just to be sure, I am running a 64 bit system, that does not change the 16?

**Comment by liskin** [comment]
Is only the rotated one laggy? Is the DP-3 okay? If so, it's definitely not the problem I had in mind, as that one was the XShmGetImage being slow, and that would affect both displays. Also, if DP-3 is okay and only the rotated VGA-0 is laggy, it's likely not a problem of screenclone.

**Comment by FortranFlo** [comment]
Yes, only the rotated display is laggy (and without rotation everything is fine). Funny enough, after experimenting, I see that only actions like dragging a window with the mouse or resizing are (very) laggy. If e.g. an xterm scrolls with lots of information, that does not lag at all. 

To clarify: if I drag (with the mouse) a window so that both monitors show part of it, the non-rotated monitor shows smooth movements, and the rotated one does not update until I release the window or pause the movement. Without rotation, all is fine.

**Comment by liskin** [comment]
That's seriously crazy. :-)
